src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -406,6 +404,11 @@ void ImageReplayer<I>::handle_bootstrap(int r) {
     on_start_fail(0, "resync requested");
     return;
   }
+  // Check if the start has been canceled only if no other
+  // bootstrap processing is required.
+  if (on_start_interrupted()) {
+    return;
+  }
```

Comment:
Why not this be part of the above `if else block`  like

```
  } else if (on_start_interrupted()) {
    return;
  }
```

Reply:
ignore! Got it from the commit msg.

Reply:
Hold on! each of the condition has a return, so this can still be part of the previous `if else` ?

Reply:
IMO, it doesn't really fit with the return value checks.  The behaviour is the same even with a separate if block.

Reply:
Yes exactly the behavior is same! hence why the change of split it into a separate block?